crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -45,6 +46,95 @@ pub trait WorldQuery {
     type State: FetchState;
 }
 
+/// # Derive
```

Comment:
Above this, I would like to see a bit of explanation on how `Fetch` works, and why end users may want to impl this at all. Ditto to `FilterFetch` obviously.

Unless you *really* dig into the engine source, their use is very opaque right now.

Reply:
I might need some help with explaining the details of both `Fetch` and` FilterFetch`. I think I can come up with explanations of why a user might need to implement it.. They are mainly about ergonomics: accessing query members as named fields, working around tuple limits, etc. But I'm not sure what to write about the internals of `Fetch`. I don't really know how archetypes and tables work under the hood, for instance, if that's what you meant you'd like to see in the docs.

Reply:
Yeah, I'll see if I can whip up some basic docs to add to this PR. It's very rough to tell users to derive a trait without any indication of what that trait is doing :)

Reply:
I've just updated the documentation, mostly writing about the details on how the derive works and explaining to users why they may want to use it. Happy to add more if you have anything else :)